server/sonar-server/src/main/java/org/sonar/server/notification/ws/ListAction.java

```diff
@@ -146,35 +150,53 @@ private boolean isDispatcherAuthorized(PropertyDto prop, String dispatcher) {
   }
 
   private Map<Long, ComponentDto> searchProjects(DbSession dbSession, List<PropertyDto> properties) {
-    Collection<String> authorizedComponentUuids = dbClient.authorizationDao().selectAuthorizedRootProjectsUuids(dbSession, userSession.getUserId(), UserRole.USER);
-    return dbClient.componentDao().selectByIds(dbSession,
-      properties.stream()
-        .filter(prop -> prop.getResourceId() != null)
-        .map(PropertyDto::getResourceId)
-        .distinct()
-        .collect(toList()))
+    Set<String> authorizedComponentUuids = new HashSet<>(dbClient.authorizationDao().selectAuthorizedRootProjectsUuids(dbSession, userSession.getUserId(), UserRole.USER));
+    Set<Long> componentIds = properties.stream()
+      .filter(prop -> prop.getResourceId() != null)
+      .map(PropertyDto::getResourceId)
```

Comment:
could also be written `.map(PropertyDto::getResourceId).filter(Objects::nonNull)` to remove the condition `!=null` :-)